gcloud/dns/changes.py

```diff
@@ -0,0 +1,261 @@
+            for added_res in resource.pop('additions', ())])
+        self._deletions = tuple([
+            ResourceRecordSet.from_api_repr(added_res, self.zone)
+            for added_res in resource.pop('deletions', ())])
+        self._properties = resource
+
+    @property
+    def path(self):
+        """URL path for change set APIs.
+
+        :rtype: string
+        :returns: the path based on project, zone, and change set names.
+        """
+        return '/projects/%s/managedZones/%s/changes/%s' % (
+            self.zone.project, self.zone.name, self.name)
+
+    @property
+    def name(self):
+        """Name of the change set.
+
+        :rtype: string or ``NoneType``
+        :returns: Name, as set by the back-end, or None.
+        """
+        return self._properties.get('id')
+
+    @name.setter
+    def name(self, value):
+        """Update name of the change set.
+
+        :type value: string
+        :param value: New name for the changeset.
+        """
+        if not isinstance(value, six.string_types):
+            raise ValueError("Pass a string")
+        self._properties['id'] = value
+
+    @property
+    def status(self):
+        """Status of the change set.
+
+        :rtype: string or ``NoneType``
+        :returns: Status, as set by the back-end, or None.
+        """
+        return self._properties.get('status')
+
+    @property
+    def started(self):
+        """Time when the change set was started.
+
+        :rtype: ``datetime.datetime`` or ``NoneType``
+        :returns: Time, as set by the back-end, or None.
+        """
+        stamp = self._properties.get('startTime')
+        if stamp is not None:
+            return datetime.datetime.strptime(stamp, _RFC3339_MICROS).replace(
+                tzinfo=UTC)
+
+    @property
+    def additions(self):
+        """Resource record sets to be added to the zone.
+
+        :rtype: sequence of
+                :class:`gcloud.dns.resource_record_set.ResourceRecordSet'.
+        :returns: record sets appended via :meth:`add_record_set`
+        """
+        return self._additions
+
+    @property
+    def deletions(self):
+        """Resource record sets to be deleted from the zone.
+
+        :rtype: sequence of
+                :class:`gcloud.dns.resource_record_set.ResourceRecordSet'.
+        :returns: record sets appended via :meth:`delete_record_set`
+        """
+        return self._deletions
+
+    def add_record_set(self, record_set):
+        """Append a record set to the 'additions' for the change set.
+
+        :type record_set:
+            :class:`gcloud.dns.resource_record_set.ResourceRecordSet'
+        :param record_set: the record set to append
+
+        :raises: ``ValueError`` if ``record_set`` is not of the required type.
+        """
+        if not isinstance(record_set, ResourceRecordSet):
+            raise ValueError("Pass a ResourceRecordSet")
+        self._additions += (record_set,)
+
+    def delete_record_set(self, record_set):
```

Comment:
Should there be some kind of concept of `committed`? I.e. deletions could be hanging around locally and never get sent off to server